okhttp/src/main/java/okhttp3/CacheControl.java

```diff
@@ -350,5 +342,11 @@ public Builder noTransform() {
     public CacheControl build() {
       return new CacheControl(this);
     }
+
+    private int getMaxValueInSeconds(int valueToConvert, TimeUnit timeUnit) {
+      long valueInSeconds = timeUnit.toSeconds(valueToConvert);
+      return valueInSeconds > Integer.MAX_VALUE
+              ? Integer.MAX_VALUE : (int) valueInSeconds;
```

Comment:
`return Math.max(Integer.MAX_VALUE, (int) valueInSeconds);` if touching this code anyway?